### DISCUSSION TITLE
Responsive design not working on Tailwind Grid

### DISCUSSION THREAD

**Discussion by srleom** [tailwindlabs/tailwindcss#12494] [Help]
Tailwind is not working when I try to use different grid layouts depending on screen size. At screen size xs, it resizes to `grid-cols-2` but does not resize to `grid-cols-3` at screen size md. 

When inspecting, it seems the `grid-cols-3` property is deleted.

<img width="381" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/32868724/169d027e-d9b4-4a7a-80c3-5a0bfb4d446e">

Below is my code. I am using the latest version of TailwindCSS v3.3.5. Appreciate any help on this, thank you!

```
<div class="flex flex-grow flex-col">
	<header class="md:w-2/3">
		<h1 class="text-4xl font-bold tracking-tight sm:text-5xl">Projects I've worked on</h1>
		<p class="mt-4 md:w-5/6">
			A gallery of projects I've built. Some of them are open-source, so if you see something that
			piques your interest, check out the code and let me know how it can be improved.
		</p>
	</header>
	<div class="mt-16 grid grid-cols-1 gap-x-12 gap-y-16 md:grid-cols-3 xs:grid-cols-2">
		<div class="">
			<Project
				img="/toDo.png"
				title="toDo"
				description="A simple toDo app with authentication functionality. Built with SvelteKit, TailwindCSS and Supabase."
				href="https://www.sveltetodo.xyz"
			/>
		</div>
		<div class=""><Project /></div>
		<div class=""><Project /></div>
		<div class=""><Project /></div>
	</div>
</div>
```

**Comment by srleom** [comment]
It seems it has something to do with my use of a custom screen size `xm`, which I added to Tailwind by extending in `tailwind.config.js`. When using standard screen sizes, it works as designed.

**Comment by Zehad-10** [comment]
tailwind css uses `min-width` for media query. I don't think your `grid-cols-3` property is deleted. you can see this `grid-cols-3` after 768px viewport width

**Comment by d0nda** [comment]
would be great if you had a reproduction to test whats really wrong

**Comment by savchukartur** [comment]
according to documentation we can't use `extend` with smaller breakpoints

<img width="350" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/58446432/9ec632a9-cdc8-4fe5-bcd9-cae15a0c96bc">

https://tailwindcss.com/docs/screens#adding-smaller-breakpoints
